src/internal_events/file.rs

```diff
@@ -82,61 +82,90 @@ mod source {
     use std::{io::Error, path::Path, time::Duration};
 
     use file_source::FileSourceInternalEvents;
-    use metrics::counter;
+    use metrics::{counter, register_counter, Counter, SharedString};
 
     use super::{FileOpen, InternalEvent};
     use crate::emit;
-    use vector_common::internal_event::{error_stage, error_type};
+    use vector_common::internal_event::{
+        error_stage, error_type, ByteSize, InternalEventHandle, RegisterInternalEvent,
+    };
 
     #[derive(Debug)]
-    pub struct FileBytesReceived<'a> {
-        pub byte_size: usize,
-        pub file: &'a str,
+    pub struct FileBytesReceived {
+        pub file: SharedString,
     }
 
-    impl<'a> InternalEvent for FileBytesReceived<'a> {
-        fn emit(self) {
-            trace!(
-                message = "Bytes received.",
-                byte_size = %self.byte_size,
-                protocol = "file",
-                file = %self.file,
-            );
-            counter!(
-                "component_received_bytes_total", self.byte_size as u64,
-                "protocol" => "file",
-                "file" => self.file.to_owned()
-            );
+    impl RegisterInternalEvent for FileBytesReceived {
+        type Handle = FileBytesReceivedHandle;
+
+        fn register(self) -> Self::Handle {
+            FileBytesReceivedHandle {
+                received_bytes: register_counter!("component_received_bytes_total", "protocol" => "file", "file" => self.file.clone()),
+                file: self.file,
+            }
         }
     }
 
-    #[derive(Debug)]
-    pub struct FileEventsReceived<'a> {
-        pub count: usize,
-        pub file: &'a str,
+    #[derive(Clone)]
+    pub struct FileBytesReceivedHandle {
+        received_bytes: Counter,
+        file: SharedString,
+    }
+
+    impl InternalEventHandle for FileBytesReceivedHandle {
+        type Data = ByteSize;
+
+        fn emit(&self, data: Self::Data) {
+            self.received_bytes.increment(data.0 as u64);
+            trace!(message = "Bytes received.", byte_size = %data.0, protocol = "file", file = %self.file);
+        }
+    }
+
+    pub struct FileEventsReceivedData {
         pub byte_size: usize,
+        pub count: u64,
     }
```

Comment:
We have a `CountByteSize` struct to handle these exact fields.

Reply:
Fixed.